src/modules/upower.cpp

```diff
@@ -379,6 +381,23 @@ void UPower::setDisplayDevice() {
           }
         },
         this);
+  } else {  // if `nativePath_` is empty, but `model_` is not.
```

Comment:
The current approach has already got the logic to filter necessary devices.  In order to have a general approach and make future extensions more flexible(such as filtering by model/path), it's better to implement a separate filtering function in order to avoid code duplication.